File: SoftUni Lessons/Python Development/Python Basics April 2019/Past Exams and Problems/9 - 20 and 21 April 2019/04. Easter Eggs Battle.py
num_eggs_first = int(input())
num_eggs_second = int(input())
command = input()
while command != "End of battle":

    if command == "one":
        num_eggs_second -= 1
    elif command == "two":
        num_eggs_first -= 1

    if num_eggs_first == 0:
        print(f"Player one is out of eggs. Player two has {num_eggs_second} eggs left.")
        break
    elif num_eggs_second == 0:
        print(f"Player two is out of eggs. Player one has {num_eggs_first} eggs left.")
        break
    command = input()
if command == "End of battle":
    print(f"Player one has {num_eggs_first} eggs left.")
    print(f"Player two has {num_eggs_second} eggs left.")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/21. МИНАЛИ ИЗПИТИ/01. Online Exam - 20 and 21 April 2019/04. Easter Eggs Battle.js
function easterEggsBattle(input) {
    // Mask
    let playerOne = Number(input.shift());
    let playerTwo = Number(input.shift());
    
    while ("someoneWinsORCommandToEnd") {
        let command = input.shift();    
        
        if (command == "End of battle") {
            console.log(`Player one has ${playerOne} eggs left.`);
            console.log(`Player two has ${playerTwo} eggs left.`);
            break;
        } else if (command == "one") {
            playerTwo -= 1;
            if (playerTwo == 0) {
                console.log(`Player two is out of eggs. Player one has ${playerOne} eggs left.`);        
                break;
            }
        } else if (command == "two") {
            playerOne -= 1;
            if (playerOne == 0) {
                console.log(`Player one is out of eggs. Player two has ${playerTwo} eggs left.`);
                break;
            }
        }
    }
}

easterEggsBattle([5, 4, 'one', 'two', 'one', 'two', 'two', 'End of battle']);  // Should return:
// Player one has 2 eggs left.
// Player two has 2 eggs left.

easterEggsBattle([2, 6, 'one', 'two', 'two']);  // Should return:
// Player two is out of eggs. Player one has 4 eggs left.